File: backend/app/schemas/notification.py
"""Notification API schemas."""

from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel

from app.schemas.common import PaginatedResponse


class NotificationResponse(BaseModel):
    id: UUID
    notification_type: str
    title: str
    message: str
    severity: str | None = None
    channel: str
    delivery_status: str
    is_read: bool
    incident_id: UUID | None = None
    created_at: datetime
    read_at: datetime | None = None


class NotificationListResponse(PaginatedResponse[NotificationResponse]):
    unread_count: int = 0

File: frontend/src/types/notification.ts
export interface AppNotification {
  id: string;
  notification_type: string;
  title: string;
  message: string;
  severity: string | null;
  channel: string;
  delivery_status: string;
  is_read: boolean;
  incident_id: string | null;
  created_at: string;
  read_at: string | null;
}

export interface NotificationListResponse {
  items: AppNotification[];
  page: number;
  page_size: number;
  total_items: number;
  total_pages: number;
  unread_count: number;
}
